word_counter.rb

```diff
@@ -0,0 +1,28 @@
+class WordCounter
+  attr_reader :filename, :word_hash
+
+  def initialize(filename)
+    @filename = filename
+    @word_hash = generate_sorted_word_hash
+  end
+
+  def count
+    @word_hash.each { |key, value| puts "#{key}: #{value}"}
+  end
+
+  def generate_sorted_word_hash
+    unsorted = Hash.new(0)
+    words = generate_word_array
+    words.each { |word| unsorted[word] += 1 }
```

Comment:
You could actually cut out a middle-man here; you define a variable `words = generate_word_array` and then enumerate over it, when you could just use `generate_word_array.each ...`. If you feel that having the words variable defined increases clarity, though, it might be worth keeping it.